core/common/src/test/scala/net/liftweb/common/BoxSpec.scala

```diff
@@ -358,7 +392,29 @@ class BoxSpec extends Specification with ScalaCheck with BoxGenerator {
       Failure("error", Empty, Empty) flatMap {x: String => Full(x.toString)} must_== Failure("error", Empty, Empty)
       Failure("error", Empty, Empty).flatten must_== Failure("error", Empty, Empty)
     }
-    "return a itself when asked for its status with the operator ?~" in {
+    "define a 'mapFailure' method that transforms it into another Failure instance." in {
+      val exception = new Exception("transformed")
+      Failure("error", Empty, Empty) mapFailure { _ => Failure("new-error", Full(exception), Empty) } must_== Failure("new-error", Full(exception), Empty)
+    }
+    "define a 'collect' method returning itself" in {
+      Failure("error", Empty, Empty) collect { case _ => "Some Value" } must_== Failure("error", Empty, Empty)
+    }
+    "define a 'collectFailure' method that takes a PartialFunction to transform this Failure into something else" in {
+      "If the partial-function is defined for this Failure, returns a full box containing the result of applying the partial function to it" in {
+        Failure("The Phantom Menace") collectFailure { case Failure("The Phantom Menace", Empty, Empty) => "Return Of The Jedi" } must_== Full("Return Of The Jedi")
+        Failure("The Phantom Menace") collectFailure { case Failure("The Phantom Menace", Empty, Empty) => Failure("Attack") } must_== Full(Failure("Attack"))
```

Comment:
So a random interesting observation here tying into your question about making `flip` take a `PartialFunction`… Now that we've added `flip`, `collectFailure` is really just a special subcase of `flip`… Do we want to keep both? Or do we, as you initially suggested, make `flip` take a `PartialFunction`, and drop `collectFailure`?

Also of interest is that `mapFailure` and `collectFailure` work very differently despite their very similar names (probably the reason why initially you defined `collectFailure` as `PartialFunction[Failure,Failure]` instead of `PartialFunction[Failure,B]`).

Not sure what the best way forward is, but I'm slightly concerned about these two things… Would be interested in thoughts/ideas before we commit to this setup as it currently stands. Going with `mapFailure` + `flip` seems like a good way to handle the common use cases without adding the oddity of `collectFailure`.